### DISCUSSION TITLE
Is there any way to forward request to another controller ?

### DISCUSSION THREAD

**Discussion by jean-smaug** [adonisjs/core#3773] [Help]
Hello,
I've got a controller in my application and when I call the update method I would like to forward request to another controller ([like in Symfony](https://symfony.com/doc/current/controller/forwarding.html)).

What is the good way to do it ?

Should I : 
- call redirect at the end of the method ?
- Use an http client like got in order to call the DELETE endpoint
- something else ?

```js
export default class RegistrationsController {
  
  public async update(ctx: HttpContextContract) {
    // Processing...
    
    // I can't make it work like that
    // ctx.response.redirect().toRoute('registrations.destroy', { id })
  }

  public async destroy(ctx: HttpContextContract) {
    // Clear
  }
}
```

Thanks 😄

**Comment by jean-smaug** [comment]
Another proposal.
Calling `await this.destroy(ctx)` at the end of the controller. 🤷‍♂️ 

Is this an anti pattern for Adonis ?

**Comment by davidharting** [comment]
I would recommend creating a different small class that contains the business logic. Your controller can then use that class to execute whatever logic is needed. The logic that is shared between the two controllers can simply be a function on that service class.

**Comment by thetutlage** [comment]
What's the use case for this?
